src/components/v2/Layout/Header/CoinInfo.tsx

```diff
@@ -0,0 +1,34 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import commaNumber from 'comma-number';
+import Typography from '@mui/material/Typography';
+import { Paper } from '@mui/material';
+import { useMarketsUser } from 'hooks/useMarketsUser';
+import { useVaiUser } from 'hooks/useVaiUser';
+import { Icon } from '../../Icon';
+import { useStyles } from './styles';
+
+const format = commaNumber.bindWith(',', '.');
+
+export const CoinInfo = ({ balance, coin }: { balance: string; coin: 'xvs' | 'vai' }) => {
+  const styles = useStyles();
+
+  return (
+    <Paper css={styles.coinInfo}>
+      <Icon name={coin} size="20px" />
+      <Typography variant="small1" noWrap component="p">
+        {balance} {coin.toUpperCase()}
+      </Typography>
+    </Paper>
+  );
+};
+
+export const XVSCoinInfo = () => {
+  const { userXVSBalance } = useMarketsUser();
+  return <CoinInfo balance={format(userXVSBalance.dp(2, 1).toString(10))} coin="xvs" />;
+};
+
+export const VAICoinInfo = () => {
+  const { userVaiBalance } = useVaiUser();
+  return <CoinInfo balance={format(userVaiBalance.dp(2, 1).toString(10))} coin="vai" />;
+};
```

Comment:
These components should be kept pure, or they'll be considered containers.

We started discussing that but I don't think we came to an actual conclusion. What we could do is consider containers as components that handle the side effect and can also compose components themselves. What I mean by this is that containers could then not only handle data fetching, calls to contracts etc.. but also hold style and render multiple components/containers.

Doing that means we could have small containers, such as here for the header, that could be nested inside larger containers such as the layout here.
This way we also keep our `components` folder free of side effect.

Reply:
This was the idea that these could be containers

Reply:
Ok got you. Yeah I'm onboard with that.
In this case I say let's just move these to the `containers` folder and we're good to go.